pkg/volume/azure_file/azure_provision.go

```diff
@@ -224,11 +225,11 @@ func (a *azureFileProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 }
 
 // Return cloud provider
-func getAzureCloudProvider(cloudProvider cloudprovider.Interface) (azureCloudProvider, error) {
+func getAzureCloudProvider(cloudProvider cloudprovider.Interface) (azureCloudProvider, string, error) {
 	azureCloudProvider, ok := cloudProvider.(*azure.Cloud)
 	if !ok || azureCloudProvider == nil {
-		return nil, fmt.Errorf("Failed to get Azure Cloud Provider. GetCloudProvider returned %v instead", cloudProvider)
+		return nil, "", fmt.Errorf("Failed to get Azure Cloud Provider. GetCloudProvider returned %v instead", cloudProvider)
 	}
 
-	return azureCloudProvider, nil
+	return azureCloudProvider, azureCloudProvider.ResourceGroup, nil
```

Comment:
this is restricting to use node resource group, while storage account could also be in external resource group, .e.g.
```
---
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: azurefile
provisioner: kubernetes.io/azure-file
parameters:
  skuName: Standard_LRS
  resourceGroup: EXISTING_RESOURCE_GROUP_NAME
```

Reply:
and in this PR, `f.resourceGroup` could only have one determined value, which means azure file share can only exists in node resource group, this issue could be the blocker here.

Reply:
one workaround is add a new field(`azurestorageresourcegroupname`) in secret:
https://github.com/kubernetes/kubernetes/blob/0c3c2cd6ac8c9ffefc38f9746034e546331b9cd6/pkg/volume/azure_file/azure_util.go#L62-L67

Reply:
I use another workaround. Set the resourceGroup into PV annotation. This way won't make users change the usage. PTAL.